src/main/java/de/tum/in/www1/artemis/web/rest/FileResource.java

```diff
@@ -130,6 +130,10 @@ public ResponseEntity<String> saveFile(@RequestParam(value = "file") MultipartFi
     @EnforceAtLeastTutor
     public ResponseEntity<byte[]> getTempFile(@PathVariable String filename) {
         log.debug("REST request to get file : {}", filename);
+        String sanitizedFileName = fileService.removeIllegalCharacters(filename);
+        if (!sanitizedFileName.equals(filename)) {
+            throw new EntityNotFoundException("Filename is invalid");
+        }
```

Comment:
Please extract this code into a method. You pasted the same code 10 times or something. Maybe use a name like ``throwIfFileNameInvalid(String filename)``

Reply:
I also asked him about this, he said this PR is a temporary solution and the next one will improve the code.

Reply:
The PR would be smaller when moving this into a method, I agree that we should create a method for it